Sync v2-7-stable with v2-7-test to release 2.7.0b1

ephraimbuddy opened the pull request:
Time for `2.7.0b1`

ephraimbuddy committed:
Update version to 2.7.0

ephraimbuddy committed:
Update RELEASE_NOTES.rst for 2.7.0

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+- Mark Task Groups as Success/Failure (#30478)
+- Add CLI command to list the provider trigger info (#30822)
+- Add Fail Fast feature for DAGs (#29406)
+- Adding configuration to control retry parameters for k8s api client (#29809)
```
This is listed under 2.6.0 is it a wrong entry here or did it miss 2.6.0?

ephraimbuddy replied:
Yeah. Getting changelog for minor releases is always a pain. I will remove that

eladkal reviewed (commented):
General comment about the release notes..
It contains many K8s and Celery changes that when merged were part of core but now they have already been released with last Providers wave. I think having them listed in core release will be very confusing to users.

WDYT?

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+- Improve the views module a bit (#31661)
+- Remove ``asynctest`` (#31664)
+- Refactor sqlalchemy queries to ``2.0`` style (#31569, #31772, #32350, #32339, #32474, #32645)
+- Remove Python ``3.7`` support (#30963)
```
I think dropping Python 3.7 should be under significant changes.
It's very important note to many users.

potiuk replied:
Yeah. I think we should also mention that the reference ``apache/airflow`` image is 3.8.

potiuk replied:
This should be done in `docs/docker-stack/changelog.rst`

ephraimbuddy replied:
Done

pierrejeambrun committed:
OpenAPI Spec fix nullable alongside $ref (#32887)

* OpenAPI Spec fix nullable alongside $ref

* Fix CI

* Update following code review

* Add deprecation warning for 'none' state.

(cherry picked from commit 3141d4bff5fb80267b3f38955838eac89c9098fd)

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+
+Disable default allowing the testing of connections in UI, API and CLI(#32052)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+The test connection functionality is disabled by default across Airflow UI,
```
```suggestion
For security reasons, the test connection functionality is disabled by default across Airflow UI,
```

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+3. ``Hidden``: Disables the test connection functionality and
+hides the Test Connection button in UI.
+For more information on capabilities of users, see the documentation:
+https://airflow.apache.org/docs/apache-airflow/stable/security/index.html#capabilities-of-authenticated-ui-users
```
This has now moved to a different URL. I think it should be:


https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html#capabilities-of-authenticated-ui-users

potiuk replied:
Tested it with locally built docs.

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+hides the Test Connection button in UI.
+For more information on capabilities of users, see the documentation:
+https://airflow.apache.org/docs/apache-airflow/stable/security/index.html#capabilities-of-authenticated-ui-users
+
```
```suggestion
It is strongly advised to **not** enable the feature until you make sure that only 
highly trusted UI/API users have "edit connection" permissions.
```

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,258 @@
+* for Celery executors you need to install ``apache-airflow-providers-celery`` package >= 3.3.0
+* for Kubernetes executors you need to install ``apache-airflow-providers-cncf-kubernetes`` package >= 7.4.0
+* For Dask executors you need to install ``apache-airflow-providers-daskexecutor`` package in any version
+
```
```suggestion
You can achieve it also by installing airflow with ``[celery]``, ``[cncf.kubernetes]``, ``[daskexecutor]`` extras respectively.

Users who base their images on the ``apache/airflow`` reference image (not slim) should be unaffected - the base
reference image comes with all the three providers installed.
```

bbovenzi committed:
Remove old graph (#32958)

(cherry picked from commit 154deed02ecc0be36b1ddfc9144aa05fb77b5f2d)

pgagnon committed:
Fix typo in PROVIDERS.rst (#33056)

Thy -> The

(cherry picked from commit 05f47b0fe87511842f894e538ddb77027bad3c55)

jkramer-ginkgo committed:
Add a note to ``Signicant Changes`` in 2.6.3 RELEASE_NOTES.rst (#33047)

(cherry picked from commit 8bfad056d8ef481cc44288c5749fa5c54efadeaa)

bbovenzi committed:
Only load task action modal if user can edit (#32992)

(cherry picked from commit 943b97850a1e82e4da22e8489c4ede958a42213d)

MatthieuBlais committed:
Add support for arrays of different data types in the Trigger Form UI (#32734)

* Add JSON field for complex arrays with types other than string

---------

Co-authored-by: Matthieu Blais <matthieu.blais@tech.jago.com>
Co-authored-by: Jens Scheffler <95105677+jens-scheffler-bosch@users.noreply.github.com>
Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
(cherry picked from commit da6931561f59a64f6489c56a5675c2cc6e6a2bca)

amoghrajesh committed:
Affix webserver access_denied warning to be configurable (#33022)

* Affix webserver access_denied warning to be configurable

(cherry picked from commit 12a760f6df831c1d53d035e4d169a69887e8bb26)

tirkarthi committed:
Update graph nodes with operator style attributes. (#32822)

* Update graph nodes with task style attributes.

* Change operator text to a tag w/ colors

---------

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>
(cherry picked from commit 3bb63f1087176b24e9dc8f4cc51cf44ce9986d34)

pankajkoti committed:
Use string concatenation for prepending base URL for log_url (#33063)

It is observed that urljoin is not yielding expected results for
the task instance's log_url which needs to be a concatenation of the
webserver base_url and specified relative url. The current usage
of urljoin does not seem to be the right way to achieve this based
on what urljoin is meant for and how it works. So, we use simple
string concatenation to yield the desired result.
More context in the comment https://github.com/apache/airflow/pull/31833#discussion_r1282696916

closes: #32996
(cherry picked from commit baa1bc0438baa05d358b236eec3c343438d8d53c)

bbovenzi committed:
remove legacy task instance modal (#33060)

(cherry picked from commit b92370fe4d96ad0fb317779fc06ef865a22c40dd)

bbovenzi committed:
Remove legacy dag code (#33058)

* Remove legacy dag code and redirect to grid view

* update docs image

(cherry picked from commit 5c384e12dea898b0bf8dee44df115e33942686a6)

vijayasarathib committed:
Documentation change to enhance readability (#33006)

(cherry picked from commit 1ff33b800246fdbfa7aebe548055409d64307f46)

ephraimbuddy committed:
Fix issue with using the various state enum value in logs (#33065)

* Fix issue with using the various state enum value in logs

The secrets masker is unable to work on the various state enums: DagRunState, TaskInstanceState,
JobState, and State enums in logs.
This PR fixes this by converting the enums to strings during the secrets mask search

* fixup! Fix issue with using the various state enum value in logs

* apply to all enum

* test with custom enum

(cherry picked from commit b0f61be2f9791b75da3bca0bc30fdbb88e1e0a8a)

potiuk committed:
Allows to choose SSL context for SMTP connection (#33070)

This change add two options to choose from when SSL SMTP connection
is created:

* default - for balance between compatibility and security
* none - in case compatibility with existing infrastructure is
  preferred

(cherry picked from commit 120efc186556b1e9498f90ad436c74e5f4e138e9)

rohan472000 committed:
Update audit_logs.rst (#33073)

Audit Logs can be found under dropdown of Browse tab in airflow UI not under Admin tab.

(cherry picked from commit 079bff9f495d57eb3564c3a22d17cbfbfba1f3c3)

potiuk committed:
Add better diagnostics when tests fail (#33023)

The "post_test" action had not worked as intended when there was
a test failure and did not upload dump of the container logs. It turned
out that the "if" conditions in the sub-action do not propagate from the
parent action, so "failure()" condition was never met.

This PR spplits post-test actions to "success" and "failure" ones, and
triggers each action respectively when the tests succeed or fail.

It also includes two fixes that makes CI debugging better:

* only logs from failed test suite are includeded in the dump
  (previously all currently running containers were included)

* we need to give some time for logs to propagate in case of errors.

(cherry picked from commit f97d1fbe1621b09bc36d2b713dde7399e2283335)

vincbeck committed:
Move security manager override to its own module (#32987)

(cherry picked from commit c10f67b36ad101a5f447a5dc075a02fffec7d0f0)

potiuk committed:
Avoid upgrade-to-newer-dependencies when provider.yaml changes (#33082)

The "upgrade-to-newer-dependencies" has been triggered when
provider.yaml changed, but this was completely unnecessary. The
real reason for upgrade is only when
generated/provider_dependencies.json changes - because that is
what changes dependencies used.

The provider_dependencies.json is automatically updated during
pre-commit based on provider.yaml files, so only using it to
trigger --upgrade-to-newer-dependencies is enough.

(cherry picked from commit 825a818eb556e8d265f3b04b9476be55a77c9689)

hussein-awala committed:
Fix BaseOperator get_task_instances query (#33054)

* Fix BaseOperator get_task_instances query

* add unit test

(cherry picked from commit f5a83bc90b237228db1434662f9dba5ebb719d47)

mobuchowski committed:
openlineage: remove eager test for auth headers, fix example dag (#33087)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>
(cherry picked from commit 1c7472df70e12f89ee18ef62b84da74386d50f90)

eumiro committed:
Get rid of Python2 numeric relics (#33050)

(cherry picked from commit e3d82c6be0e0e1468ade053c37690aa1e0e4882d)

uranusjr committed:
Ensure DAG-level references are filled on unmap (#33083)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit bcfadcf6e4b2de587959594f54a9e8fef96c4a2b)

potiuk committed:
Fix test failing after open-lineage 1.0 API change (#33095)

One more test fix needed after #33087

(cherry picked from commit 153e489dd02d1ca182e7eef65acdcb5225802a9b)

potiuk committed:
Temporarily exclude openlineage from "PyPI" constraints generation (#33101)

The openlineage provider has apache-airflow >= 2.7.0 as dependency,
and this is a problem for automated constraint generation. Since there
is only one version of the provider released, it's the only one that
can be installed from PyPI, but until apache-airflow 2.7.0 is released
it cannot be installed because of the dependency missing.

This makes calculating of the constraints by PyPI impossible and it
loops in continuously trying to find a solution that cannot be found.

In order to unblock constraint generation we need to exclude the
provider temporarily and regenerate the constraints again when airflow
2.7.0 gets released (which will generally only bring the open-lineage
provider as part of the constraints).

(cherry picked from commit b08188e24bf7e52a0b700e083caa38347a4fa3fc)
(cherry picked from commit f14430d4d34413ea7aae0378a5169e4a16597555)

dstandish committed:
Use arrow with no circle for setup / teardown (#33098)

Just seems easier to see.

(cherry picked from commit b57fc45861a7527541e684c00333d983290a88c0)

dstandish committed:
Don't auto-add to context just by virtue of arrowing (#33102)

* Don't auto-add to context just by virtue of arrowing

* no add tasks

* Add back removed import

* Fixup tests

* fixup! Fixup tests

---------

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit c18a5a9f8630d7b9ec464eb30d61f084d01ffcec)

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,296 @@
+https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html#capabilities-of-authenticated-ui-users
+It is strongly advised to **not** enable the feature until you make sure that only
+highly trusted UI/API users have "edit connection" permissions.
+The ``xcomEntries`` API disables support for the ``deserialize`` flag by default (#32176)
```
Empty line here?

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,296 @@
+""""""""""""""""""""""""""""""""""
+The old Graph View is removed. The new Graph View is the default view now.
+
+In case of SMTP SSL connection, the default context now uses "default" context (#33070)
```
In case of SMTP SSL connection, the context now uses "default" context (#33070)

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,296 @@
+
+In case of SMTP SSL connection, the default context now uses "default" context (#33070)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+The "default" context is Python's ``default_ssl_contest`` instead of previously used "none". The
```
I think those paragraphs here would benefit from a little more spacing in general. This looks too much like a wall of text, splitting it into paragraphs and separating them with empty lines would be much more readable.

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,296 @@
+deserialization at the client side instead.
+
+Default name of the Celery application changed from ``airflow.executors.celery_executor`` to ``airflow.providers.celery.executors.celery_executor`` (#32526)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
This likely should be first paragraph - but the header should be simpler: 

``Change of the celery app name``  ?

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,296 @@
+To keep the behavior as close as possible to the old config, one can set ``scheduler.max_tis_per_query = 0``,
+in which case it'll always use the value of ``core.parallelism``.
+
+The Kubernetes, Celery, CeleryKubernetes, LocalKubernetes, and Dask executors are moved to corresponding providers (#32767)
```
Same here. I think the header should be:

``Some executors have been moved to providers`` or smth like that.

potiuk reviewed (commented):
Few NIT's when it comes to readability of the release notes.
